lib/LaTeXML/Engine/TeX_Box.pool.ltxml

```diff
@@ -337,11 +337,12 @@ Tag('svg:foreignObject', autoOpen => 1, autoClose => 1,
     # Make sure we get a size, in case autoOpen'd
     if ($whatsit) {
       my ($w, $h, $d) = $whatsit->getSize;
-      my $y  = $STATE->lookupDefinition(T_CS('\baselineskip'))->valueOf->pxValue;
-      my $ht = $h->add($d);
-      $node->setAttribute(width     => $w->pxValue)  unless $node->hasAttribute('width');
-      $node->setAttribute(height    => $ht->pxValue) unless $node->hasAttribute('height');
-      $node->setAttribute(transform => "matrix(1 0 0 -1 0 $y)");
+      my $ht = $node->getAttribute('height') // $h->add($d)->pxValue;
+      $h = $h->pxValue;
+      $node->setAttribute(style     => "line-height:${h}px");
+      $node->setAttribute(width     => $w->pxValue) unless $node->hasAttribute('width');
+      $node->setAttribute(height    => $ht)         unless $node->hasAttribute('height');
+      $node->setAttribute(transform => "matrix(1 0 0 -1 0 $h)");
```

Comment:
This took a few tries. Shifting by height only mimics better the behaviour of the depth, but it's not enough on its own. The additional `line-height` ensures that boxes with same horizontal coordinate and font size get the same baseline (this is noticeable when an arrow has multiple labels in tikzcd).

Reply:
Quite possibly (or even probably) recording `@rowdepths` is a good thing, but it would be useful to have in its own PR, without the flex & div wrappers (I'd like to explore alternatives for that).

Reply:
Correction: `line-height` is just a mirage. I got good results using `text-box` which is very new CSS, not yet supported by Firefox. It might resolve that 'maths too high' issue you were mentioning.

I couldn't find any other way of aligning the blocks by *baseline*, unfortunately. I suggest we use `text-box` as the reference starting point and once it looks convincing, work on a fallback for older browsers.